src/builder.rs

```diff
@@ -347,6 +351,62 @@ impl<'ctx> Builder<'ctx> {
         PointerValue::new(value)
     }
 
+    /// Alignment arguments are specified in bytes, and should always be a power of 2.
+    ///
+    /// The final argument should be a pointer-sized integer.
+    ///
+    /// [`TargetData::ptr_sized_int_type_in_context`](https://thedan64.github.io/inkwell/inkwell/targets/struct.TargetData.html#method.ptr_sized_int_type_in_context) will get you one of those.
+    #[llvm_versions(8.0..=latest)]
+    pub fn build_memcpy(
+        &self,
+        dest: PointerValue<'ctx>,
+        dest_align_bytes: u32,
+        src: PointerValue<'ctx>,
+        src_align_bytes: u32,
+        size: IntValue<'ctx>,
+    ) -> Result<PointerValue<'ctx>, &'static str> {
```

Comment:
Correct me if I'm wrong, but this function cannot fail for the moment. Why returning `Result` then? Is it because it's a WIP 🙂 ?

Reply:
Yeah, because it's WIP - I was thinking about having it return `Err` if either alignment value isn't a power of 2, but it turns out if you provide something that isn't a power of 2, LLVM fails module verification anyway.

Based on that, I'm thinking this (and `build_memmove`) can just return `PointerValue<'ctx>` instead of a `Result`.

What do you think?